Tests/Shared.Specs/CollectionEquivalencySpecs.cs

```diff
@@ -1142,6 +1142,46 @@ public void When_asserting_equivalence_of_non_generic_collections_it_should_resp
                 .WithMessage("*Wheels*not have*VehicleId*not have*");
         }
 
+        [Fact]
+        public void When_comparing_against_a_non_generic_collection_it_should_treat_it_as_unordered_collection_of_objects()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            List<Type> actual = new List<Type> { typeof(int), typeof(string) };
+            IEnumerable expectation = new List<Type> { typeof(string), typeof(int) };
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => actual.Should().BeEquivalentTo(expectation);
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_comparing_against_a_non_generic_collection_it_should_treat_it_as_collection_of_objects()
```

Comment:
Oh crap. My plan was to add another test that would use that overload indeed. Thanks for being my conscience during this weekend ;-)

Reply:
Without the extra overloads, that new test actually ends up going to the overload that takes a `params object[]`

Reply:
Apparently everyone wants to be a `params object[]`.
This PR would break a test such as:

```c#
// Arrange
var actual = new[] { new[] { typeof(int), typeof(string) } };
IEnumerable expectation = new List<Type> { typeof(string), typeof(int) };

// Act
Action act = () => actual.Should().BeEquivalentTo(expectation);

// Assert
act.Should().NotThrow();
```

Reply:
I'm still working on it (in-between my duties), but it ended up being more complicated than I thought ;-)

Reply:
When the expectation is a non-generic collection, FA is supposed to use the run-time types of the objects. This was already the case. I've just made it more explicit.

Reply:
I'm not sure I follow. 
Does this mean that the changes here are still going to break the test above?
No matter what, I think the test should be added to act as documentation for the intended behavior.

Reply:
Yes, it will, since that's the expected behavior. You are comparing a collection that contains one item (the collection containing two `Type`s) with a collection that contains two items (the two `Type`s). So unless I'm reading something wrong, I expect this to fail.

Reply:
@jnyrup ?